pkg/storage/sqlite/sqlite_test.go

```diff
@@ -0,0 +1,358 @@
+
+			if tt.mixed {
+				curTuple, err = iter.Head(ctx)
+				require.NoError(t, err)
+				require.Equal(t, thirdTuple, curTuple.GetKey())
+			}
+
+			curTuple, err = iter.Next(ctx)
+			require.NoError(t, err)
+			require.Equal(t, thirdTuple, curTuple.GetKey())
+
+			if tt.mixed {
+				_, err = iter.Head(ctx)
+				require.ErrorIs(t, err, storage.ErrIteratorDone)
+			}
+
+			_, err = iter.Next(ctx)
+			require.ErrorIs(t, err, storage.ErrIteratorDone)
+		})
+	}
+}
+
+// TestReadPageEnsureNoOrder asserts that the read page is ordered by ulid.
+func TestReadPageEnsureOrder(t *testing.T) {
+	testDatastore := storagefixtures.RunDatastoreTestContainer(t, "sqlite")
+
+	uri := testDatastore.GetConnectionURI(true)
+	ds, err := New(uri, sqlcommon.NewConfig())
+	require.NoError(t, err)
+	defer ds.Close()
+
+	ctx := context.Background()
+
+	store := "store"
+	firstTuple := tuple.NewTupleKey("doc:object_id_1", "relation", "user:user_1")
+	secondTuple := tuple.NewTupleKey("doc:object_id_2", "relation", "user:user_2")
+
+	err = sqlcommon.Write(ctx,
+		sqlcommon.NewDBInfo(ds.db, ds.stbl, sq.Expr("datetime('subsec')")),
+		store,
+		[]*openfgav1.TupleKeyWithoutCondition{},
+		[]*openfgav1.TupleKey{firstTuple},
+		time.Now())
+	require.NoError(t, err)
+
+	// Tweak time so that ULID is smaller.
+	err = sqlcommon.Write(ctx,
+		sqlcommon.NewDBInfo(ds.db, ds.stbl, sq.Expr("datetime('subsec')")),
+		store,
+		[]*openfgav1.TupleKeyWithoutCondition{},
+		[]*openfgav1.TupleKey{secondTuple},
+		time.Now().Add(time.Minute*-1))
+	require.NoError(t, err)
+
+	opts := storage.ReadPageOptions{
+		Pagination: storage.NewPaginationOptions(storage.DefaultPageSize, ""),
+	}
+	tuples, _, err := ds.ReadPage(ctx,
+		store,
+		tuple.NewTupleKey("doc:", "relation", ""),
+		opts)
+	require.NoError(t, err)
+
+	require.Len(t, tuples, 2)
+	// We expect that objectID2 will return first because it has a smaller ulid.
+	require.Equal(t, secondTuple, tuples[0].GetKey())
+	require.Equal(t, firstTuple, tuples[1].GetKey())
+}
+
+func TestReadAuthorizationModelUnmarshallError(t *testing.T) {
+	testDatastore := storagefixtures.RunDatastoreTestContainer(t, "sqlite")
+
+	uri := testDatastore.GetConnectionURI(true)
+	ds, err := New(uri, sqlcommon.NewConfig())
+	require.NoError(t, err)
+
+	ctx := context.Background()
+	defer ds.Close()
+	store := "store"
+	modelID := "foo"
+	schemaVersion := typesystem.SchemaVersion1_0
+
+	bytes, err := proto.Marshal(&openfgav1.TypeDefinition{Type: "document"})
+	require.NoError(t, err)
+	pbdata := []byte{0x01, 0x02, 0x03}
+
+	_, err = ds.db.ExecContext(ctx, "INSERT INTO authorization_model (store, authorization_model_id, schema_version, type, type_definition, serialized_protobuf) VALUES (?, ?, ?, ?, ?, ?)", store, modelID, schemaVersion, "document", bytes, pbdata)
+	require.NoError(t, err)
+
+	_, err = ds.ReadAuthorizationModel(ctx, store, modelID)
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "cannot parse invalid wire-format data")
+}
```

Comment:
We don't need this test :) @jpadilla can correct me if i'm wrong, but I think this was only applicable to MySQL and Postgres whilst in the middle of the migration from "one record per type" to "one record per model"

Reply:
Yep can drop this for new storage implementations